src/app/public/modules/datepicker/datepicker.component.ts

```diff
@@ -31,6 +31,8 @@ import {
   SkyDatepickerCalendarComponent
 } from './datepicker-calendar.component';
 
+let nextId = 0;
```

Comment:
There were a few `datepicker` things that slipped through the cracks in the previous PR. I've fixed them here since they were small. The rest of the PR should be `timepicker` specific.